microsoft-edge/webview2/concepts/windowed-vs-visual-hosting.md

```diff
@@ -1,436 +1,194 @@
+<!-- ------------------------------ -->
+#### APIs for Window to Visual hosting
 
----
+For a list of APIs that can be used when configuring WebView2 Window to Visual hosting (or for Windowed hosting), see [Rendering WebView2 in non-framework apps](./overview-features-apis.md#rendering-webview2-in-non-framework-apps) in _Overview of WebView2 features and APIs_.
 
 
 <!-- ====================================================================== -->
-## Visual hosting
-
-In visual hosting, content is embedded to a given location on the application. This location must handle how content will scale and behave in an app when the user interacts with the application.  In addition to the window management described for windowed hosting, visual hosting will need the app to manage the composition-based rendering, when it receives any user interactions.  For more information about visual hosting, see [Using the Visual layer in desktop apps](/windows/apps/desktop/modernize/visual-layer-in-desktop-apps).
-
-If your WebView2 app uses visual hosting:
+## Visual hosting: For more granular control over layout
 
-* Inputs are routed to the app's `HWND` and must be configured to send the spatial input (for example, mouse, touch, and pen) based on positions, _not_ what currently has focus, such as a keyboard.
-
-The following APIs can be used when configuring WebView2 in a visual hosting environment:
+When using _Visual hosting_, your host app receives spatial input (such as mouse or touch input) from the user, and forwards this input to the WebView2 control.  Visual hosting requires the app to do the same window management as Windowed hosting, but has additional window management requirements regarding:
+* Scaling the contents.
+* Routing spatial inputs (such as mouse, touch, or pen).
 
 
 <!-- ------------------------------ -->
-#### Composition-based rendering
-
-For composition-based WebView2 rendering, use the `CoreWebView2Environment` to create a `CoreWebView2CompositionController`. The `CoreWebView2CompositionController` implements all the APIs as `CoreWebView2Controller`, but includes additional APIs specific to composition-based rendering.
-
-##### [.NET/C#](#tab/dotnetcsharp)
-
-* [CoreWebView2CompositionController Class](/dotnet/api/microsoft.web.webview2.core.corewebview2compositioncontroller)
-* [CoreWebView2Environment.CreateCoreWebView2CompositionControllerAsync Method](/dotnet/api/microsoft.web.webview2.core.corewebview2environment.createcorewebview2compositioncontrollerasync)
+#### Requirements for scaling the contents
 
-##### [WinRT/C#](#tab/winrtcsharp)
+Per composition-based rendering, a WebView2 control is part of a Visual tree, so by default, it's rendered at the scale of its ancestor Visual.  For example, if a WebView2's ancestor Visual is scaled 2x, then the WebView2's contents are also rendered at 2x scale.  But because the WebView2 is not scaling its own contents, they're blurry.
```

Comment:
Reading this again - I'm a little wary of the wording. I had it worded more ambiguously before because the WebView2 will be affected by the scales of all of its ancestor visuals. This wording implies only 1 visual will affect the WebView2. But for instance, if the WebView2's parent visual is zoomed 2x and that visual's parent is zoomed 2x, then the WebView2 would be zoomed 4x. WinComp/Dcomp developers should know this which is why I wasn't this explicit in my explanation.

Reply:
result:
```
Per composition-based rendering, a WebView2 control is part of a Visual tree, so by default, it's rendered at a scale that's based on the scales of all of its ancestor Visuals.  For example, if a WebView2's ancestor Visual is scaled (zoomed) 2x, then the WebView2's contents are also rendered at 2x scale.  If the WebView2's parent Visual is scaled 2x and that Visual's parent is also scaled 2x, then the WebView2 is scaled 4x.  But because the WebView2 is not scaling its own contents, they're blurry.
```